src/field/RungeKuttaStepper.hh

```diff
@@ -0,0 +1,83 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file RungeKuttaStepper.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "base/Range.hh"
+#include "base/Types.hh"
+#include "base/Macros.hh"
+//#include "FieldEquation.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * The 4th order classical Runge-Kutta stepper
+ *
+ * This method estimates the updated state from an initial state and slope
+ * estimate, with fourth-order accuracy.
+ *
+ * For a magnetic field equation \em f with state \em x, which includes
+ * position and momentum but may also include time and spin:
+ * \f[
+ *  XXX TODO: document steps here in latex math
+ * \f]
+ */
+template<class FieldEquation_T>
+class RungeKuttaStepper
+{
+    //@{
+    //! Type aliases
+    using OdeState = typename FieldEquation_T::OdeState;
+    //@}
+
+  public:
+    //! Construct with the equation of motion
+    inline CELER_FUNCTION RungeKuttaStepper(FieldEquation_T& equation);
```

Comment:
The equation shouldn't need to be mutable, should it? The stepping shouldn't change the equation or its state.

Reply:
Added const.